src/main/scala/com/fulcrumgenomics/FgBioDef.scala

```diff
@@ -25,8 +25,23 @@
 package com.fulcrumgenomics
 
 import com.fulcrumgenomics.commons.CommonsDef
+import enumeratum.{Enum, EnumEntry}
 
 /** FgBioDef that is just an extension of the commons CommonsDef. Here in case
   * there's a need for FgBio specific defs later.
   */
-object FgBioDef extends CommonsDef
+object FgBioDef extends CommonsDef {
+  /** Each case object in your enumeration should extend this trait. */
+  trait ClpEnumEntry extends EnumEntry
+
+  /** Extends this trait in your enumeration object to enable the case objects to be created on the command line.
+    * You should implement the [[Enum#values]] method in your object, for example:
+    * `def values: IndexedSeq[T] = findValues`.
+    * */
+  trait ClpEnum[T<:ClpEnumEntry] extends Enum[T] {
+    // NB: we cannot put `def values: IndexedSeq[T] = findValues` here since ClpEnum is not a class!
+
+    /** The string constructor method that enables us to create the case object instance by name. */
+    def apply(name: String): T = withName(name)
```

Comment:
It sorta sucks that we have to do this here.  I would have thought that Enumeratum would supply an `apply()` method :/